### PR TITLE
copr: Vectorize left shift

### PR DESCRIPTION
**Pull request by waynexia**
PCP #5751 

###  What have you changed?

Vectorize Left Shift

###  What is the type of the changes?

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?
Please select the tests that you ran to verify your changes:
- Unit test
- Integration test
###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No
###  Does this PR affect `tidb-ansible`?
No

### CODE REVIEW

**Comment by waynexia** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/31 tidb=pr/13443

File: components/tidb_query/src/rpn_expr/impl_op.rs [hunk 1]
```diff
@@ -97,6 +97,15 @@ fn decimal_is_false(arg: &Option<Decimal>) -> Result<Option<i64>> {
     Ok(Some(arg.as_ref().map_or(0, |v| v.is_zero() as i64)))
 }
 
+#[rpn_fn]
+#[inline]
+fn left_shift(arg0: &Option<i64>, arg1: &Option<i64>) -> Result<Option<i64>> {
+    Ok(match (arg0, arg1) {
+        (Some(arg0), Some(arg1)) => Some((*arg0 as i64).checked_shl(*arg1 as u32).unwrap_or(0)),
```
Comment (andylokandy) [inline components/tidb_query/src/rpn_expr/impl_op.rs]: `arg0` should be cast to `u64`

File: components/tidb_query/src/rpn_expr/impl_op.rs [hunk 2]
```diff
@@ -343,4 +352,26 @@ mod tests {
             assert_eq!(output, expect_output, "{:?}, {:?}", arg, sig);
         }
     }
+
+    #[test]
+    fn test_left_shift() {
+        let test_cases = vec![
```
Comment (andylokandy) [inline components/tidb_query/src/rpn_expr/impl_op.rs]: Suggest to add these test cases: 

- `None, None => None`
- `Some(123), Some(2) => Some(492)` 
- `Some(-123), Some(-1) => Some(0)`

File: components/tidb_query/src/rpn_expr/impl_op.rs (hunk 1 above)
Comment (andylokandy) [reply components/tidb_query/src/rpn_expr/impl_op.rs]: The result should be cast to `i64`

**Comment by andylokandy** [comment]
I'm sorry, https://github.com/tikv/tikv/pull/5886 has been merged. 😢

**Comment by waynexia** [comment]
Alright, thanks.

**Pull request closed without merging**